### PR TITLE
[SATURN-1884] Uploader page

### PR DESCRIPTION
**Pull request by andrewmarcus**
This PR implements a new feature allowing state labs to easily upload their entity data and create new tables in their workspace. Much of this functionality should eventually be ported back into the Workspaces view.

Amongst the recent updates:

* The abort upload button now works. I see that it can sometimes get into a state where the upload has stopped but the modal isn't dismissable, which I think happens because when the actual fetch is aborted, an unresolved promise is returned. But I didn't want to change the plumbing of the deep-down AJAX code. In the general case the button is working.
* I've switched out the UI to more of an accordion style rather than the nav buttons on the left side. After putting it in place, this interface seems MUCH more fluid and intuitive. See what you think.

I have not yet managed to resolve the issue with the `AutoSizer`. Please have your UX team take a look at it and tell me what I did wrong.

### CODE REVIEW

**Comment by andrewmarcus** [comment]
I just got some accessibility improvements in to the Uploader feature, some of which improve the site generally.

* Allow spacebar to select buttons (generally)
* Added aria-modal='true' to modals (generally)
* Show hover states on focus (generally)
* Move focus to section header when it opens, to help the user track their place (Uploads)
* Read out the current filter state to screen readers (Uploads > workspace)

I also hid the button for creating new folders for new

File: src/components/Dropzone.js [hunk 1]
```diff
@@ -25,10 +30,18 @@ const Dropzone = ({ disabled = false, onDragOver, onDrop, onDragLeave, style = {
...
+    },
+    onDropRejected: (...args) => {
+      setRejected(true)
+      onDropRejected && onDropRejected(...args)
+    },
     ...props
   })
 
-  return div(getRootProps({ tabIndex: -1, style: { ...style, ...(dragging ? activeStyle : {}) } }), [
+  return div(getRootProps({ tabIndex: -1, style: { ...style, ...(dragging ? activeStyle : {}), ...(rejected ? rejectedStyle : {}) } }), [
```
Comment (dmohs) [inline src/components/Dropzone.js]: Not seeing usage of `rejectedStyle` or `fileRejections` anywhere.

File: src/components/Interactive.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ const Interactive = Utils.forwardRefWithName('Interactive', ({
     className: `hover-style ${className}`,
     style: { ...style, ...cssVariables, cursor: computedCursor, outline },
     onKeyDown: onKeyDown || (e => {
-      if (e.key === 'Enter') {
+      if (e.key === 'Enter' || e.key === ' ') {
```
Comment (dmohs) [inline src/components/Interactive.js]: I think this makes sense, but should be a separate PR unless essential.

File: src/components/Modal.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ const Modal = ({ onDismiss, title, titleExtras, children, width = 450, showCance
       nodeToFocus.focus()
     },
     style: { overlay: styles.overlay, content: { ...styles.modal, width } },
-    aria: { labelledby: titleId },
+    aria: { labelledby: titleId, modal: true },
```
Comment (dmohs) [inline src/components/Modal.js]: This is a great change, but also a separate PR unless essential to this functionality.

File: src/components/Modal.js [hunk 4]
```diff
@@ -74,9 +74,9 @@ const Modal = ({ onDismiss, title, titleExtras, children, width = 450, showCance
         }, [cancelText]) :
         null,
       Utils.cond(
-        [okButton === undefined, () => h(ButtonPrimary, { onClick: onDismiss }, 'OK')],
-        [_.isString(okButton), () => h(ButtonPrimary, { onClick: onDismiss }, okButton)],
-        [_.isFunction(okButton), () => h(ButtonPrimary, { onClick: okButton }, 'OK')],
+        [okButton === undefined, () => h(ButtonPrimary, { onClick: onDismiss, danger }, 'OK')],
+        [_.isString(okButton), () => h(ButtonPrimary, { onClick: onDismiss, danger }, okButton)],
+        [_.isFunction(okButton), () => h(ButtonPrimary, { onClick: okButton, danger }, 'OK')],
```
Comment (dmohs) [inline src/components/Modal.js:79]: I like this, but consider whether it can be omitted for the sake of preserving existing files as much as possible (when introducing new functionality). This would be a great addition as a follow-up PR.

File: src/components/table.js [hunk 5]
```diff
@@ -394,25 +394,35 @@ GridTable.propTypes = {
...
         style: { display: 'flex' }, className: 'table-row',
-        hover: { backgroundColor: colors.light(0.4) }
+        hover: { backgroundColor: colors.light(0.4) },
+        role: 'row'
       }, [
         _.map(({ key, size }) => {
           return div({
             key,
             className: 'table-cell',
+            role: 'gridcell',
```
Comment (dmohs) [inline src/components/table.js]: Love these role additions, but as a separate PR so we can contain the new functionality as much as possible.

File: src/style.css [hunk 6]
```diff
@@ -7,7 +7,7 @@
 /* This adds some low-level css resets based on the browserslist in package.json, see https://create-react-app.dev/docs/adding-css-reset */
 @import-normalize;
 
-.hover-style:hover {
+.hover-style:hover, .hover-style:focus {
```
Comment (dmohs) [inline src/style.css]: Separate PR? I think we want these kinds of changes unconditionally, whereas the upload feature has some organizational controversy around it.

**Review by dmohs — COMMENTED** [review]
This first round is focused on keeping the changed set contained as much as possible. In general, I'm impressed with how well you follow the existing style.

File: src/components/Dropzone.js (hunk 1 above)
Comment (andrewmarcus) [reply src/components/Dropzone.js]: Thanks for the feedback. I'd be happy to pull out the more general accessibility changes into separate PRs. There are a couple of things in here, such as this change to Dropzone that I was going to use but then ended up doing something else. They might still be useful in the future but I'll pull them out for this pass.

File: src/components/Modal.js (hunk 4 above)
Comment (andrewmarcus) [reply src/components/Modal.js:79]: This particular change allowed me to make the Abort Upload button red instead of blue, which more accurately captures its purpose (don't press that button unless you really mean it). However, I'd be happy to pull that out into a different PR for cleanliness.

File: src/components/Modal.js (hunk 3 above)
Comment (andrewmarcus) [reply src/components/Modal.js]: Sure. I'll pull this and the other changes out into a separate PR handling the first pass at accessibility improvements. These particular roles and keyboard actions impact the accessiblity of the new uploads page, which is why I included them, but there are no particular dependencies that these be in there for the first pass.

File: src/style.css (hunk 6 above)
Comment (andrewmarcus) [reply src/style.css]: Thanks for that, good to know. Happy to split it up

File: src/components/Modal.js (hunk 4 above)
Comment (dmohs) [reply src/components/Modal.js:79]: Tough call, probably should stay in this PR.

File: src/components/FloatingActionButton.js [hunk 7]
```diff
@@ -23,7 +23,8 @@ const FloatingActionButton = ({ label, iconShape, onClick, bottom = 30, right =
       onClick: () => {
         onClick()
         setHover(false)
-      }
+      },
+      ...props
```
Comment (dmohs) [inline src/components/FloatingActionButton.js]: This isn't right, I think. The ref you're passing here is creating a warning. I believe you should be able to find the right pattern elsewhere in the codebase.

Comment (andrewmarcus) [reply src/components/FloatingActionButton.js]: In the FloatingActionButton? I'm not sure I understand which ref you're referring to. The only thing I changed in this file was to pass through additional optional properties to the button. That said, I don't recall now whether I actually ended up needing the extra props here. I'll double-check and remove them if I can.

Comment (dmohs) [reply src/components/FloatingActionButton.js]: It's used to pass a ref on line 683 of Upload.js. I'd say leave that out until we can get it right. We can do a usability pass later.

Comment (dmohs) [reply src/components/FloatingActionButton.js]: (the ref is used to focus a button)

**Comment by andrewmarcus** [comment]
I got the table AutoSizer working. The problem turned out to be that I just hadn't been passing down the `height: 100%` appropriately through all the flexbox layers.

**Comment by dmohs** [comment]
This is excellent. Changes are well-contained to new files, additions to existing files, and very few changes to existing code, where the existing style is respected. I judge this is safe to merge and we can follow-up with improvements in later PRs. I will work on getting a second approval.

**Comment by dmohs** [comment]
Tested manually by running through the demonstrated workflow. Only minor cosmetic and usability issues found, nothing that should block release.

**Review by dmohs — APPROVED** [review]

**Review by abaumann — APPROVED** [review]
Reviewed the functionality and intended flow by running this branch locally and it all looks correct to me according to the requirements
